Updated database handling and html parsing and increased integration for plugins

CreaoticX opened the pull request:
This borrows heavily from Wordpress in order to increase the ability for plugins to integrate with SEO Panel.  It does this primarily by adding the ability to incorporate action and filter hooks into the SEO Panel operations and by adding the ability save meta data associated with any data type, but specifically websites, projects, and reports.  

Functions to simplify managing settings were also added.  Settings can now include arrays and can be excluded from being made constants by adding the prefix 'ign_' to the category.

In the process of adding this functionality,  an html parser and a MySQLI database handler were also added to simplify operations, increase readability, and increase security.  A mysqli helper file was also added to maintain backwards compatibility until the existing code can be

sendtogeo committed:
Update README

sendtogeo committed:
Delete README

sendtogeo committed:
Create README.md

Commit:
Seo Panel 3.7.0

szepeviktor committed:
Update 3.6.0 -> 3.7.0

szepeviktor committed:
reset $supportClass in adminmenu.ctp.php×2

sendtogeo committed:
Merge pull request #1 from szepeviktor/version_num

Update 3.6.0 -> 3.7.0

sendtogeo committed:
Merge pull request #4 from szepeviktor/reset-supportClass

Reset $supportClass in adminmenu.ctp.php×2

Commit:
seopanel.v.3.8.0 - Changes to fix the database text values short tag replace issue and validation bug fix

Commit:
Merge branch 'master' of https://github.com/seopanel/Seo-Panel

szepeviktor committed:
Merge cron's separate debug property with SP_DEBUG

Commit:
v.3.8.0 - Adding changes for payment plugin

Commit:
adding git ignore file

Commit:
.git ignore file changes

Commit:
seo panel 3.8.0 - Adding currency related changes

Commit:
seo panel 3.8.0 - adding .gitignore changes

Commit:
seo panel 3.8.0 - currency settings changes

Commit:
seo panel 3.8.0 - currency settings changes

Commit:
seo panel 3.8.0: To fix the forgot password button not showing error

sendtogeo committed:
Merge pull request #6 from szepeviktor/merge-cron-debug

Merge cron's separate debug property with SP_DEBUG

sendtogeo committed:
Update cron.ctrl.php

Commit:
seo panel 3.8.0 - Adding website import feature

Commit:
seo panel 3.8.0 - Adding the ssl bypass fix and header provding option

drao-aqorn committed:
Test

gvarghese123 committed:
add tmp folder

gvarghese123 committed:
add tmp folder

gvarghese123 committed:
new table for user specs

sendtogeo committed:
new table for user specs

deepthyraor committed:
User type changes

Commit:
seo panel 3.8.0 - website import feature

Commit:
seo panel 3.8.0 - minor fixes

deepthyraor committed:
User type edits

Commit:
seo panel 3.8.0 - adding pagination to dashboard

Commit:
seo panel 3.8.0 - user expiry date setup

deepthyraor committed:
Changes for adding website / keyword count checking for user types

Commit:
seo panel 3.8.0 - changes to filters

Commit:
seo panel 3.8.0 - to get default user id and admin id

sendtogeo committed:
seo panel 3.8.0 - The changes for subscription

sendtogeo committed:
seo panel 3.8.0 : changes for pagination in keyword report summary

sendtogeo committed:
seo panel 3.8.0 : changes for pagination in keyword report summary

sendtogeo committed:
seo panel 3.8.0 : changes for pagination in archive report summary

sendtogeo committed:
seo panel 3.8.0 : changes for pagination in archive report summary

Commit:
seo panel 3.8.0 : subscription changes

Commit:
seo panel 3.8.0 : subscription changes

Commit:
seo panel 3.8.0 - small cosmetic changes

Commit:
seo panel 3.8.0 - subscription module changes

Commit:
seo panel 3.8.0 - subscription module changes

Commit:
seo panel 3.8.0 - subscription module changes

Commit:
seo panel 3.8.0 - cosmetic changes

Commit:
seo panel 3.8.0 - cosmetic changes

Commit:
seo panel 3.8.0 - cosmetic changes

Commit:
seo panel 3.8.0 - copying changes to simple theme

Commit:
seo panel 3.8.0 - copying changes to simple theme

Commit:
seo panel 3.8.0 - cosmetic changes

sendtogeo committed:
seo panel 3.8.0 - cosmetic changes

Commit:
seo panel 3.8.0 - cosmetic changes

Commit:
seo panel 3.8.0 - cosmetic changes

Commit:
seo panel 3.8.0 - cosmetic changes

Commit:
seo panel 3.8.0 - cosmetic changes

Commit:
seo panel 3.8.0 - cosmetic changes

Commit:
seo panel 3.8.0 - cosmetic changes

sendtogeo committed:
Seo panel 3.8.0 - Adding all changes from develop

sendtogeo committed:
seo panel demo changes

sendtogeo committed:
Merge pull request #23 from seopanel/develop

Develop

Commit:
seo panel 3.9.0 - cosmetic changes

Commit:
seo panel 3.8.1 -  changes to fix the https link detection issue

Commit:
seo panel 3.8.1 - To fix teh issue with reciprocal filter

Commit:
seo panel 3.9.0 - cosmetic changes

Commit:
seo panel 3.9.0: to fix the php tag issues

Commit:
reverting config file changes

sendtogeo committed:
fix issue #35

sendtogeo committed:
fix issue #34

sendtogeo committed:
fix issue #41

sendtogeo committed:
fix issue  #26

sendtogeo committed:
3.9.0 changes

sendtogeo committed:
fix issue #13

sendtogeo committed:
fix issue #38

sendtogeo committed:
fix issue #40

sendtogeo committed:
3.9.0 changes

Commit:
keyword report summary changes

sendtogeo committed:
keyword postion summary changes #36 #37

sendtogeo committed:
revert unwanted commit

Commit:
keyword report summary changes #36 #37

sendtogeo committed:
keyword postion summary changes #36 #37

Commit:
fixed issue #28 static function call issue

Commit:
fixed issue #27 phpmailer new version

Commit:
fixed issue #21

Commit:
fixed issue #21

Commit:
replace google pagerank with moz rank

sendtogeo committed:
replace google pagerank with moz rank #40

Commit:
changes for dashboard #44

Commit:
changes for check connection of moz rank settings

sendtogeo committed:
replace google pagerank with moz rank #40

Commit:
to fix #48

Commit:
seo panel final changes for 3.9.0

Commit:
seo panel 3.9.0 changes

Commit:
seo panel 3.9.0 changes

Commit:
to order crawl log

sendtogeo committed:
Merge pull request #49 from seopanel/develop

Develop

Commit:
seo panel 3.9.0 cosmetic changes

sendtogeo committed:
Merge pull request #50 from seopanel/develop

seo panel 3.9.0 cosmetic changes

CreaoticX committed:
Added hooks functionality

Added a hooks class that was ported from Wordpress.  This allows hooks and filters to be added to the main code so that plugins can tie into the main code.

CreaoticX committed:
Added common functions for using hooks

Creating the common functions for the hooks removes the need to call the hook global every time.

CreaoticX committed:
Added load capability for plugins

Adds the ability to include a load.php file in a plugin folder that will be loaded on start up each time.  This will allow plugins to add functionality to the core system.

CreaoticX committed:
Fixed plugin load path

Added the full path instead of just the directory to the plugin load function.

CreaoticX committed:
Altered the Bar Menu

Put the bar menu into a for loop and added a filter so that menus could be added or removed.

CreaoticX committed:
Added MySqli Handler and Wrapper

Mysqli handler allows for prepared statements for security.  A wrapper was added for backwards compatibility until all queries can be updated.

CreaoticX committed:
Add setting functions

Added common setting functions for adding, updating, and getting settings

CreaoticX committed:
Fixed get_setting_value

Changed get_setting_value to check for defined variable first

CreaoticX committed:
Added PHP HTML Parser

Adding a more robut HTML parser to allow for greater analysis of crawled pages.

CreaoticX committed:
Added meta tables and functions

Created tables for meta data and ported functions from Wordpress to use in handling meta data

CreaoticX committed:
Made formatUrl function more precise

formatUrl now removes whitespace and only removes protocols at the very beginning of the url

CreaoticX committed:
Updated getPageInfo and related functions

Updated the getPageInfo function to utilize the PHPHtmlParser and added new filters to allow for customized page info

CreaoticX committed:
Fixed str_replace error

str_replace was throwing an error a number 1 set as the limit so a $limit variable was created and inserted

CreaoticX committed:
Fixed URL formatting and added filter info

Fixed the URL removing the domain name.  Added a boolean $internal variable to the link_info filters.

CreaoticX committed:
Fixed URL formatting errors

Fixed two more errors in formatting the link url

CreaoticX committed:
Removed duplicate home page entries and changed broken link check

Eliminated duplicate home page issue for domain ending in / and changed the broken link checker to check the database before pinging the website.

CreaoticX committed:
Added Update Setting Function

Added update setting function to simplify updating settings without having to enter all the setting data.

CreaoticX committed:
Updated Settings to Allow For Arrays

Added maybe serialize functions to saving and getting settings functions

CreaoticX committed:
Added ability to prevent settings from being defined

By adding the prefix ign_ to the category of a setting it will not be loaded as a constant

CreaoticX committed:
Updated database handler and added filters to auditor component

Updated the database calls to use the new MySQLi database handler and added filters and action hooks to the scoring functions

CreaoticX committed:
Merge branch 'develop' of https://github.com/seopanel/Seo-Panel

# Conflicts:
#	controllers/components/auditorcomponent.php
#	includes/sp-common.php
#	includes/sp-load.php
#	libs/seopanel.class.php
#	libs/spider.class.php
#	themes/classic/views/menu/adminmenu.ctp.php
#	themes/classic/views/menu/usermenu.ctp.php

sendtogeo commented:
Hi Team,

Thanks a lot for these changes send by you.

We will verify and will add it to seo panel repository and will update you.

Thansk for the support

CreaoticX committed:
Created dynamic scoring and editting of scoring settings

Separated scoring functions into separate classes to allow for additional scoring factors to be added dynamically and factor weighting and settings to be editted by the user.

TODO:
Create score weight and setting editting pages
Create domain scoring factors

CreaoticX committed:
Fixed scoring algorithm

Divided the total score by the total of all weights and multiplied by 10 to give a total score between -10 and 10

CreaoticX committed:
fixed untracked files

CreaoticX committed:
Reverted Classic theme

Reverted the classic theme to it's original form.  Creating a new theme to make changes in instead

CreaoticX committed:
Load Scores

Added include to load the the default score classes

CreaoticX committed:
Fixed Score Broken Link File name

CreaoticX committed:
Check for non-array passed to validate settings

CreaoticX committed:
Changed Score Class variables for Private to Protected

This makes them accessible by the Score Class children

CreaoticX committed:
Added default label to Score Settings

A label is required to save a setting.

CreaoticX committed:
Added Active state to Score Class

Scores can now be activated and deactivated

CreaoticX committed:
Removed Deactivated Scores From scoring algorithm

CreaoticX committed:
Moved Active check in Score Class

Active check was after default return and wasn't being called.

CreaoticX committed:
Check for runkit_constant_redefine function

If runkit_constant_redefine function does not exist don't try to reset constants

CreaoticX committed:
Bug Fixed Save Settings

Fixed bug causing updated score settings not to save properly

CreaoticX committed:
Added Website Average Page Score

Added a score to handle the average page score for websites.

TODO:
Implement website scores

CreaoticX committed:
Separate functions for calculating score and weighted score and added dynamic scoring for projects

The countScore and weightedScore functions can now be used for any score type.  These are now used to calculate the score for projects as well.

CreaoticX committed:
Load new project score for pages average score

sendtogeo commented:
Thanks again for the support by writing this much code.

Can you please send the merge request to following develop branch of seo panel

https://github.com/seopanel/Seo-Panel/tree/develop

CreaoticX committed:
Merge pull request #1 from seopanel/master

Updates from https://github.com/seopanel/Seo-Panel/
